src/main/java/htsjdk/samtools/BAMShardIndexer.java

```diff
@@ -0,0 +1,190 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.Log;
+
+import java.io.File;
+import java.io.OutputStream;
+
+public class BAMShardIndexer {
+  // output written as binary, or (for debugging) as text
+  private final BinaryBAMShardIndexWriter outputWriter;
+
+  // content is built up from the input bam file using this
```

Comment:
This file and others in this folder are mostly copied as is from HTSJDK - I will add comments to that effect.
But I am going to largely ignore other comments about changes in this file since its not my intent to cleanup HTSJDK code that already works and in fact I'd rather preserve the code close to the original version, warts and all ;)